File: app/src/main/java/com/xhg/test/image/MainActivity.java
package com.xhg.test.image;

import android.content.Intent;
import android.os.Bundle;
import android.support.v7.app.ActionBar;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.DefaultItemAnimator;
import android.support.v7.widget.GridLayoutManager;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.support.v7.widget.Toolbar;
import android.util.Log;
import android.view.Menu;
import android.view.MenuItem;
import android.view.View;
import android.widget.Button;
import android.widget.Toast;

import com.xhg.test.image.adapter.MainRecyclerAdapter;
import com.xhg.test.image.model.StrategyModel;
import com.xhg.test.image.settings.SettingsActivity;
import com.xhg.test.image.strategies.ColorStrategy;

import java.util.Arrays;
import java.util.List;

/**
 * @author xionghg
 * @email xiong9394@gmail.com
 * @created 2017-07-04.
 */

public class MainActivity extends AppCompatActivity {

    private static final String TAG = "TestBitmap-Main";

    private Button mToButton;
    private RecyclerView mRecyclerView;
    private List<ColorStrategy> mStrategies;
    private MainRecyclerAdapter mRecyclerAdapter;
    private StrategyModel mStrategyModel;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        Toolbar toolbar = (Toolbar) findViewById(R.id.toolbar);
        setSupportActionBar(toolbar);
        ActionBar actionBar = getSupportActionBar();
        if (actionBar != null) {
            //显示导航按钮
            actionBar.setDisplayHomeAsUpEnabled(true);
            //设置图标
            actionBar.setHomeAsUpIndicator(R.drawable.ic_menu_white_32dp);
        }

        initData();

        mToButton = (Button) findViewById(R.id.to_button);
        mToButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent intent = new Intent(MainActivity.this, BigImageActivity.class);
                intent.putExtra("strategy_index", 4);
                startActivity(intent);
            }
        });

        mRecyclerView = (RecyclerView) findViewById(R.id.recycle_view);
        LinearLayoutManager layoutManager = new GridLayoutManager(MainActivity.this, 2, GridLayoutManager.VERTICAL, false);
        //设置布局管理器
        mRecyclerView.setLayoutManager(layoutManager);
        mRecyclerAdapter = new MainRecyclerAdapter(MainActivity.this, mStrategies);
        //设置Adapter
        mRecyclerView.setAdapter(mRecyclerAdapter);
        //设置增加或删除条目的动画
        mRecyclerView.setItemAnimator(new DefaultItemAnimator());
        mRecyclerAdapter.setOnItemClickListener(new MainRecyclerAdapter.OnItemClickListener() {
            @Override
            public void onItemClick(View view, int position) {
                Log.d(TAG, "onItemClick: position=" + position);
                Intent intent = new Intent(MainActivity.this, BigImageActivity.class);
                intent.putExtra("strategy_index", position);
                startActivity(intent);
            }

            @Override
            public void onItemLongClick(View view, int position) {
            }
        });
    }

    private void initData() {
        mStrategyModel = StrategyModel.getInstance();
        mStrategies = Arrays.asList(mStrategyModel.getStrategies());
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        getMenuInflater().inflate(R.menu.toolbar, menu);
        return true;
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        switch (item.getItemId()) {
            case android.R.id.home:
                Toast.makeText(this, "Home function is coming soon", Toast.LENGTH_SHORT).show();
                break;
            case R.id.settings:
                Intent intent = new Intent(MainActivity.this, SettingsActivity.class);
                startActivity(intent);
                // Toast.makeText(this, "To be coming soon", Toast.LENGTH_SHORT).show();
                break;
            case R.id.help:
                Toast.makeText(this, "To be coming soon", Toast.LENGTH_SHORT).show();
                break;
            default:
                break;
        }
        return true;
    }
}

File: app/src/main/java/com/xhg/test/image/MainActivity.kt
package com.xhg.test.image

import android.content.Intent
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.support.v7.widget.RecyclerView
import android.support.v7.widget.Toolbar
import android.view.LayoutInflater
import android.view.MenuItem
import android.view.ViewGroup
import android.widget.TextView
import android.widget.Toast
import com.xhg.test.image.pictures.PicturesActivity
import java.util.*

class MainActivity : AppCompatActivity() {

    private var testItems: MutableList<TestItem> = ArrayList()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val toolbar = findViewById<Toolbar>(R.id.toolbar)
        setSupportActionBar(toolbar)
        supportActionBar?.let {
            // 设置图标
            it.setHomeAsUpIndicator(R.drawable.ic_menu_white_32dp)
            // 显示导航按钮
            it.setDisplayHomeAsUpEnabled(true)
        }

        val recyclerView = findViewById<RecyclerView>(R.id.recycle_view)
        getList()
        recyclerView.adapter = MainAdapter(testItems)
    }

    private fun getList() {
        testItems.add(TestItem("Image Test", PicturesActivity::class.java))
        testItems.add(TestItem("Image Test", PicturesActivity::class.java))
        testItems.add(TestItem("Image Test", PicturesActivity::class.java))
        testItems.add(TestItem("Image Test", PicturesActivity::class.java))
        testItems.add(TestItem("Image Test", PicturesActivity::class.java))
        testItems.add(TestItem("Image Test", PicturesActivity::class.java))
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        when (item.itemId) {
            android.R.id.home -> Toast.makeText(this, "Home function is coming soon", Toast.LENGTH_SHORT).show()
            else -> {
            }
        }
        return super.onOptionsItemSelected(item)
    }

    internal class TestItem(internal var name: String, internal var cls: Class<*>)

    internal class MainAdapter(var data: MutableList<TestItem>?) : RecyclerView.Adapter<MainHolder>() {

        override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): MainHolder {
            return MainHolder(parent)
        }

        override fun getItemCount(): Int {
            return data?.size ?: 0
        }

        override fun onBindViewHolder(holder: MainHolder, position: Int) {
            data?.let { holder.bindData(it[position], position) }
        }
    }

    internal class MainHolder(viewGroup: ViewGroup) : RecyclerView.ViewHolder(LayoutInflater.from(viewGroup.context).inflate(R.layout.list_item_main_test, viewGroup, false)) {

        private val textView: TextView = itemView.findViewById(R.id.test_item_name)

        fun bindData(testItem: TestItem, position: Int) {
            textView.text = testItem.name
            itemView.setOnClickListener { itemView.context.startActivity(Intent(itemView.context, testItem.cls)) }
        }
    }

}
